src/lightstep-opentelemetry-node.ts

```diff
@@ -0,0 +1,239 @@
+import {
+  ConsoleLogger,
+  LogLevel,
+  CompositePropagator,
+} from '@opentelemetry/core';
+import { HttpTextPropagator, Logger } from '@opentelemetry/api';
+import { NodeSDK } from '@opentelemetry/sdk-node';
+import {
+  LightstepConfigurationError,
+  LightstepNodeSDKConfiguration,
+  LightstepEnv,
+  LS_DEFAULTS,
+  LS_OPTION_ALIAS_MAP,
+  PropagationFormat,
+  PROPAGATION_FORMATS,
+  PROPAGATOR_LOOKUP_MAP,
+} from './types';
+import {
+  CollectorTraceExporter,
+  CollectorProtocolNode,
+} from '@opentelemetry/exporter-collector';
+import {
+  Resource,
+  ResourceLabels,
+  SERVICE_RESOURCE,
+} from '@opentelemetry/resources';
+
+let logger: Logger;
+let fail: (message: string) => void;
+
+/**
+ * Returns a NodeSDK object configured using Lightstep defaults with user
+ * overrides. Call start on the returned NodeSDK instance to apply the
+ * configuration and start the export pipeline.
+ * @param config
+ */
+export function configureOpenTelemetry(
+  config: Partial<LightstepNodeSDKConfiguration> = {}
+): NodeSDK {
+  logger = setupLogger(config);
+  fail = config.failureHandler || defaultFailureHandler(logger);
+
+  config = coalesceConfig(config);
+  validateConfiguration(config);
+  configureBaseResource(config);
+  configurePropagation(config);
+  configureTraceExporter(config);
+
+  return new NodeSDK(config);
+}
+
+/**
+ * Setup up logger to use for LOCL. This may or may not be the logger configured
+ * for OpenTelemetry. This is so we can print meaningful error messages when
+ * configuration fails.
+ */
+function setupLogger(config: Partial<LightstepNodeSDKConfiguration>): Logger {
+  if (config.logger) return config.logger;
+
+  let logLevel: LogLevel = config.logLevel ?? LogLevel.INFO;
+
+  if (process.env.OTEL_LOG_LEVEL) {
+    logLevel =
+      LogLevel[
+        process.env.OTEL_LOG_LEVEL.toUpperCase() as keyof typeof LogLevel
```

Comment:
It's a string in other languages. I wrote this so that a user can pass in `OTEL_LOG_LEVEL=debug` instead of `OTEL_LOG_LEVEL=3`. This makes the configuration consistent with other locl packages. We can discuss whether this is the right thing to do. It's easier for us to read a number, but harder for users.

Reply:
hmm fair point then should we raise a ticket and change it in otel so this will be aligned across all languages ?

Reply:
yeah, i think that would be a good idea.